platforms/documentation/docs/src/docs/userguide/reference/gradle_daemon.adoc

```diff
@@ -212,40 +215,111 @@ The following file will be generated:
+Gradle recognizes some JVM vendor strings as equivalent, such as "Adoptium" and "Temurin."
+You can view the list of recognized vendors by running `gradle help --task updateDaemonJvm`.
 
-Gradle recognizes a small number of JVM vendor strings as special and equivalent. For example, "Adoptium" and "Temurin" are considered the same vendor. You can see the list of special vendors by running `gradle help --task updateDaemonJvm`.
+If the specified vendor is not one of the recognized equivalents, Gradle will match it exactly.
+For example, "My Custom JVM" would require an exact match of the vendor name.
 
-If the JVM vendor you specify is not treated as a special value, Gradle considers the value as an exact match. For example, to match the vendor "My Custom JVM", the vendor criterion must be "My Custom JVM".
+The vendor can be configured on the task using <<toolchains.adoc#sec:vendors,the same syntax as the Java toolchain spec>>.
 
+[[sec:daemon_jvm_discovery]]
 === Daemon JVM discovery
 
 To locate a compatible JVM installation, Gradle re-uses the mechanism provided by the <<toolchains.adoc#toolchains,Java Toolchains>> feature.
 This feature is used to locate a JVM installation that matches the criteria specified in the `gradle/gradle-daemon-jvm.properties` file.
 
-NOTE: The _daemon JVM discovery process_ does not support auto-provisioning of new JVM installations.
-This will be added in a future release.
+This process fully supports both <<toolchains.adoc#sec:auto_detection,auto-discovery>> and <<toolchains.adoc#sec:provisioning,auto-provisioning>>.
+
+[[sec:daemon_jvm_provisioning]]
+==== Controlling auto-provisioning URLs
+
+By default, the `updateDaemonJvm` task attempts to generate download URLs for JDKs on various platforms (OS and architecture) that match the specified criteria.
+
+Gradle considers the following build platforms by default:
+
+* [`FREE_BSD`, `AARCH64`]
+* [`LINUX`, `AARCH64`]
+* [`LINUX`, `X86_64`]
+* [`MAC_OS`, `AARCH64`]
+* [`MAC_OS`, `X86_64`]
+* [`UNIX`, `AARCH64`]
+* [`UNIX`, `X86_64`]
+* [`WINDOWS`, `X86_64`]
+
+These platforms can be configured through the `toolchainPlatforms` property of the `UpdateDaemonJvm` task.
+Note that there is currently no CLI option for configuring this.
+
+====
+include::sample[dir="snippets/daemon-jvm/customized-task/kotlin",files="build.gradle.kts[tags=customized-plat-daemon-jvm-task]"]
+include::sample[dir="snippets/daemon-jvm/customized-task/groovy",files="build.gradle[tags=customized-plat-daemon-jvm-task]"]
+====
+
+Gradle resolves JDK download URLs for these platforms by using the <<toolchains.adoc#sub:download_repositories,configured toolchain download repositories>>.
+If no such repositories are configured, the `updateDaemonJvm` task will fail.
```

Comment:
🤔 May want to add "unless an empty list of platforms has been explicitly configured."
~~And possible add a snippet for this.~~

Since that's done already below, maybe this should be called out here.

My reasoning: I am being told something can fail, making another thing mandatory. I would like to know immediately that there are different options to "resolve" the failure.

Reply:
Fixed.